### PR TITLE
Integrate debugger stepping. Closes #212

### LINKED ISSUE #212: Implement Debug Stepping
Right now we've implemented the ensime  server's breakpoint api, but not the step api. While you can accomplish stepping with just breakpoints, we may as well support the full step api.

### PR DESCRIPTION
**Pull request by ktonga**
This is a cleanup of what @Jacoby6000 started in #284 
It needs to land before code keeps diverging after so many refactors

Add support for:

- Step
- Step-Out
- Next

### CODE REVIEW

**Comment by ches** [comment]
Needs a `make format` 😄

**Comment by ktonga** [comment]
Yeap. Done.

File: ensime_shared/debugger.py [hunk 1]
```diff
@@ -65,3 +65,21 @@ def debug_backtrace(self, args, range=None):
...
+            "threadId": self.debug_thread_id})
+
+    def debug_step_out(self, args, range=None):
+        self.log("debug_step_out: in")
+        self.send_request({
+            "typehint": "DebugStepOutReq",
+            "threadId": self.debug_thread_id})
+
+    def debug_next(self, args, range=None):
+        self.log("debug_next: in")
```
Comment (ches) [inline ensime_shared/debugger.py]: Since the logging is merged these will need to be `self.log.debug` not to break.

Comment (ktonga) [inline ensime_shared/debugger.py]: Ups, you're right.

**Comment by ches** [comment]
There we go, LGTM!

**Pull request merged**